src/rpc/misc.cpp

```diff
@@ -320,6 +320,73 @@ UniValue createmultisig(const JSONRPCRequest& request)
     return result;
 }
 
+UniValue getsignaturehash(const JSONRPCRequest& request)
+{
+    if (request.fHelp || request.params.size() < 5 || request.params.size() > 6)
+        throw std::runtime_error(
+            "getsignaturehash \"tx\" \"scriptcode\" \"amount\" \"sigversion\" inputindex (\"sighashtype\")\n"
+            "\nCreate the hash to sign for the given input\n"
+
+            "\nArguments:\n"
+            "1. \"tx\"     (string, required) The transaction hex string\n"
+            "2. \"scriptcode\"     (string, required) The scriptCode to sign\n"
```

Comment:
This doesn't explain what it is.

Reply:
the scriptCode is difficult to explain, it depends on `OP_CODESEPARATOR` and whether you are using Segwit/P2SH/Segwit-P2SH/Normal script